src/lowering/desugar.ml

```diff
@@ -142,10 +154,11 @@ and exp' at note = function
       I.PrimE (I.CastPrim (T.seq ts1, T.seq ts2), [exp e])
     | _ -> assert false
     end
-  | S.CallE ({it=S.AnnotE ({it=S.PrimE "serialize";_}, _);note;_}, _, e) ->
+  | S.CallE ({it=S.AnnotE ({it=S.PrimE "serialize";_}, _);note;_}, _,
+             {it=S.TupE es; _}) ->
```

Comment:
@crusso related to my other comment.  To make the IR type checker work for expansions of `to_candid` that become `serialize`, I tweaked the syntax and IR typing rule for `serialize` to be an n-ary expression node, like we chose for `to_candid`.